.github/workflows/nightly-codeql-analysis.yml

```
@@ -0,0 +1,50 @@
+name: Nightly CodeQL analysis
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: '30 1 * * *'
+
+jobs:
+  analyze:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Java 17
+        uses: actions/setup-java@v2
+        with:
+          distribution: temurin
+          java-version: 17
+
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v1
+        with:
+          languages: java
+
+      - uses: gradle/gradle-build-action@v2
+        with:
+          arguments: assemble
```

```suggestion
          arguments: assemble --no-build-cache
```

Should we try this in SDK repo to see if it fixes the analyzed issue?

---

oh, good idea, I will send PR